backend/icpy/agent/custom_agent.py

```
@@ -173,54 +156,53 @@ def get_configured_custom_agents() -> List[Dict[str, Any]]:
+                        try:
+                            return new_loop.run_until_complete(registry.discover_and_load())
+                        finally:
+                            new_loop.close()
+                    
+                    with concurrent.futures.ThreadPoolExecutor() as executor:
+                        future = executor.submit(init_registry)
+                        _ = future.result(timeout=5)  # 5 second timeout
+                else:
+                    # No running loop, safe to use run_until_complete
+                    _ = loop.run_until_complete(registry.discover_and_load())
+            except Exception as init_error:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)